src/JuliaInterpreter.jl

```diff
@@ -20,12 +20,22 @@ module CompiledCalls
 # This module is for handling intrinsics that must be compiled (llvmcall)
 end
 
+const BUILTIN_FILE = joinpath(@__DIR__, "builtins-julia$(Int(VERSION.major)).$(Int(VERSION.minor)).jl")
+
+if !isfile(BUILTIN_FILE)
```

Comment:
Should we also check `Base.stale_cachefile("generate_builtins.jl", cachefile)`? I'm not sure I'm thinking about the logic of this properly, but it seems to make sense to rebuild it if the generator changes.

If possible, it would be lovely to preserve the ability to edit the "builtins-" file for the purposes of debugging. If I'm thinking about it correctly, it seems that checking the mtime on "generate_builtins.jl" but *not* "builtins-" will preserve that capability.

Reply:
Storing the `mtime` of `generate_builtins.jl` and rerun the build script if it changed makes sense.

> If possible, it would be lovely to preserve the ability to edit the "builtins-" file for the purposes of debugging.

The changes here shouldn't affect that?

Reply:
Right, that's what I'm expecting too. But sometimes the chain of dependencies makes my head hurt.

Reply:
I don't think you have to store the `mtime`, do you? That comparison should be run for you by `stale_cachefile`.

Reply:
Maybe another approach: put this logic inside a `ccall(:jl_generating_output, Cint, ()) == 1` and add the `generate_builtins.jl` as an `include_dependency`?

Reply:
Something like this then?